src/SFA.DAS.EmployerFinance.MessageHandlers/EventHandlers/EmployerFinance/StartedProcessingLevyDeclarationsEventHandler.cs

```diff
@@ -0,0 +1,26 @@
+using System.Threading.Tasks;
+using MediatR;
+using NServiceBus;
+using SFA.DAS.EmployerFinance.Application.Commands.ImportLevyDeclarations;
+using SFA.DAS.EmployerFinance.Application.Commands.UpdateLevyDeclarationSagaProgress;
+using SFA.DAS.EmployerFinance.Messages.Events;
+
+namespace SFA.DAS.EmployerFinance.MessageHandlers.EventHandlers.EmployerFinance
+{
+    public class StartedProcessingLevyDeclarationsEventHandler : IHandleMessages<StartedProcessingLevyDeclarationsEvent>
+    {
+        private readonly IMediator _mediator;
+
+        public StartedProcessingLevyDeclarationsEventHandler(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        public Task Handle(StartedProcessingLevyDeclarationsEvent message, IMessageHandlerContext context)
+        {
+            return Task.WhenAll(
+                _mediator.Send(new ImportLevyDeclarationsCommand(message.SagaId)),
```

Comment:
The `StartedProcessingLevyDeclarationsAdHocEventHandler` passes the `PayrollPeriod` from the `StartedProcessingLevyDeclarationsAdHocEvent` to the `ImportPayeSchemeLevyDeclarationsCommand`.

Here, we don't pass the `PayrollPeriod` from the `StartedProcessingLevyDeclarationsEvent` to the `ImportLevyDeclarationsCommand` (it gets picked up from the saga instead).

Should we be consistent? If we end up not using `PayrollPeriod` in `StartedProcessingLevyDeclarationsEvent`, should we remove it?

Reply:
Hmm yeah we could include more data on the event. Currently we have:

* `ImportLevyDeclarationsCommand` - Requires saga lookup, just sends commands.
* `ImportPayeSchemeLevyDeclarationsCommand` - Doesn't require saga lookup, does actual work.
* `UpdateLevyDeclarationTransactionBalancesCommand` - Requires saga lookup, just sends commands.
* `UpdateAccountLevyDeclarationTransactionBalancesCommand` - Doesn't require saga lookup, does actual work.

By including everything on the event (e.g. `AccountPayeSchemeHighWaterMarkId` etc) we could avoid the saga lookups when running `ImportLevyDeclarationsCommand` & `UpdateLevyDeclarationTransactionBalancesCommand`. What do you think?

Reply:
`UpdateLevyDeclarationTransactionBalancesCommandHandler` retrieves the saga by the id supplied in the `UpdateLevyDeclarationTransactionBalancesCommand`, but then only uses the Id from the saga anyway, so it looks like we can remove the saga lookup there anyway.

As for `ImportLevyDeclarationsCommandHandler`, I think eliminating the saga lookup by including `AccountPayeSchemeHighWaterMarkId` and `PayrollPeriod` should provide us a nice little performance boost!

Reply:
Done :+1: